### PR TITLE
Update k8s.io/* dependencies to v0.26.14 and Go to 1.22

### PR DESCRIPTION
**Pull request by zaklawrencea**
This PR:
- updates k8s.io/* to v0.26.14
- updates Go to 1.22

### CODE REVIEW

File: kubernetes/pods.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func (a *Adapter) PodInformer(ctx context.Context, endpointChan chan<- []string)
 	queueEndpoints(&podEndpoints, endpointChan)
 
 	// delta triggered updates
-	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+	_, _ = informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
```
Comment (zaklawrencea) [inline kubernetes/pods.go:53]: Changed due to golangci-lint:
```bash
golangci-lint run --timeout 5m ./...
kubernetes/pods.go:53:26: Error return value of `informer.AddEventHandler` is not checked (errcheck)
        informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
                                ^
make: *** [Makefile:27: lint] Error 1
```

Comment (szuecs) [reply kubernetes/pods.go:53]: Maybe we should check the error and at least log it for now. Later we can decide to ignore or handle it

File: controller.go [hunk 2]
```diff
@@ -11,12 +11,12 @@ import (
 	"syscall"
 	"time"
 
+	kingpin "github.com/alecthomas/kingpin/v2"
 	"github.com/aws/aws-sdk-go/service/elbv2"
 	log "github.com/sirupsen/logrus"
 	"github.com/zalando-incubator/kube-ingress-aws-controller/aws"
 	"github.com/zalando-incubator/kube-ingress-aws-controller/certs"
 	"github.com/zalando-incubator/kube-ingress-aws-controller/kubernetes"
-	kingpin "gopkg.in/alecthomas/kingpin.v2"
```
[the remark this answers is not in the document]
Comment (zaklawrencea) [reply controller.go]: Sure, dropped.

File: kubernetes/pods.go (hunk 1 above)
Comment (zaklawrencea) [reply kubernetes/pods.go]: Sounds good, updated to log the error.

**Comment by szuecs** [comment]
:+1:

**Comment by zaklawrencea** [comment]
:+1:

**Pull request merged**